src/warc2zim/converter.py

```diff
@@ -319,6 +321,22 @@ def gather_information_from_warc(self):
 
             # if we get here, found record for the main page
 
+            # if main page is a redirect, update the main url accordingly
+            if record.http_headers:
+                status_code = int(record.http_headers.get_statuscode())
+                if status_code in [
```

Comment:
I think we should handle the case of multiple redirects

Reply:
Nevermind, I've found a pretty good test URL : `https://www.kiwix.org/`, which redirects to `https://www.kiwix.org/en/` and then `https://kiwix.org/en/`

Reply:
👍

Reply:
In fact browsertrix crawler is already "fully" resolving the redirect, so there will be only one redirect in all situations:

```
{"timestamp":"2024-03-01T09:35:45.493Z","logLevel":"info","context":"worker","message":"Starting page","details":{"workerid":0,"page":"http://www.kiwix.org/"}}
{"timestamp":"2024-03-01T09:35:45.494Z","logLevel":"info","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":0,"total":1,"pending":1,"failed":0,"limit":{"max":10,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2024-03-01T09:35:45.412Z\",\"extraHops\":0,\"url\":\"http://www.kiwix.org/\",\"added\":\"2024-03-01T09:35:45.031Z\",\"depth\":0}"]}}
{"timestamp":"2024-03-01T09:35:46.485Z","logLevel":"info","context":"general","message":"Awaiting page load","details":{"page":"http://www.kiwix.org/","workerid":0}}
{"timestamp":"2024-03-01T09:35:47.645Z","logLevel":"info","context":"general","message":"Seed page redirected, adding redirected seed","details":{"origUrl":"http://www.kiwix.org/","newUrl":"https://kiwix.org/en/","seedId":1}}
```

Reply:
Good to know. Should we document it in the code?

Reply:
I was misled by Browsertrix logs, there are many redirects in the WARC (and it is handled mostly nicely, just had to fix a small bug due to the fact that "Location" header might be a relative path only, not a full URL)
```
--------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------
[warc2zim::2024-03-01 11:21:07,708] WARNING:HTTP 301 occurred on main page; replacing http://www.kiwix.org/ with 'https://www.kiwix.org/'
[warc2zim::2024-03-01 11:21:07,709] WARNING:HTTP 301 occurred on main page; replacing https://www.kiwix.org/ with 'https://www.kiwix.org/en/'
[warc2zim::2024-03-01 11:21:07,709] WARNING:HTTP 301 occurred on main page; replacing https://www.kiwix.org/en/ with 'https://kiwix.org/en/'
```